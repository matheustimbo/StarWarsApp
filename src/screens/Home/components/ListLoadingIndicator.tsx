import React from 'react';
import {View, StyleSheet, ActivityIndicator} from 'react-native';

// import { Container } from './styles';

const ListLoadingIndicator: React.FC<{loadingPeople: Boolean}> = ({
  loadingPeople,
}) => {
  if (loadingPeople) {
    return (
      <View style={styles.loadingIndicator}>
        <ActivityIndicator color="black" size="large" />
      </View>
    );
  }
  return null;
};

const styles = StyleSheet.create({
  loadingIndicator: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 12,
  },
});

export default ListLoadingIndicator;
