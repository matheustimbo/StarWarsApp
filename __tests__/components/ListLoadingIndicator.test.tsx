import React from 'react';
import renderer from 'react-test-renderer';
import ListLoadingIndicator from '../../src/screens/Home/components/ListLoadingIndicator';

test('renders correctly when loading people', () => {
  const tree = renderer
    .create(<ListLoadingIndicator loadingPeople={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('return null when not loading', () => {
  const tree = renderer
    .create(<ListLoadingIndicator loadingPeople={false} />)
    .toJSON();
  expect(tree).toBe(null);
});
