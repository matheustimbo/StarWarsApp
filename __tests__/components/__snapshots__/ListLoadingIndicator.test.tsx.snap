// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when loading people 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginVertical": 12,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#fff"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
